tests/test_derivatives.py

```diff
@@ -773,13 +773,49 @@ def test_param_stagg_inner(self):
         xp = x + x.spacing / 2
 
         f = TimeFunction(name="f", grid=grid, space_order=space_order, staggered=y)
-        p = Function(name="p", grid=grid, space_order=space_order, parameter=True)
+        p = Function(name="p", grid=grid, space_order=space_order)
         g = TimeFunction(name="g", grid=grid, space_order=space_order, staggered=(x, y))
 
         eqn = Eq(g, (p * f).dx)
         eqne = eqn.evaluate.rhs
         assert simplify(eqne - (p._subs(y, yp).evaluate * f).dx(x0=xp).evaluate) == 0
 
+    def test_param_stagg_add(self):
```

Comment:
quite a few changes have been made, and it seems weird that only one new test was required?

were they all fixing the same exact (set of) thing(s), which is entirely captured by this test? if so, OK